scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -623,12 +635,6 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
   @inline final def commaSeparated[T <: Tree : AstMetadata](part: => T): List[T] =
     tokenSeparated[`,`, T](sepFirst = false, part)
```

Comment:
Can `tokenSeparated` be changed to accommodate the suggested structure of `constructR1Quasi`? It seems that it could be designed to tell the underlying `ellipsis` to skip the provided tokens if necessary. As a side-effect, we'll also get code reuse as a bonus.